src/weapons.h

```diff
@@ -0,0 +1,382 @@
+    AMMO_GRENADE_NORMAL = 39,                          // TR4+
+    AMMO_GRENADE_SUPER = 40,
+    AMMO_GRENADE_FLASH = 41,
+    AMMO_CROSSBOW_NORMAL = 44,
+    AMMO_CROSSBOW_SUPER = 46,
+    AMMO_CROSSBOW_POISON = 45,
+    AMMO_GRAPPLIN = 47
+};
+
+// all weapon sound
+enum weapon_sound_e
+{
+    SND_NULL = -1,
+    SND_UNIVERSAL_RELOAD = TR_AUDIO_SOUND_RELOAD,
+    SND_UNIVERSAL_DRAW = TR_AUDIO_SOUND_HOLSTEROUT,
+    SND_UNIVERSAL_HIDE = TR_AUDIO_SOUND_HOLSTERIN,
+    SND_PISTOL = TR_AUDIO_SOUND_SHOTPISTOLS,
+    SND_SHOTGUN = TR_AUDIO_SOUND_SHOTSHOTGUN,
+    SND_UZI = TR_AUDIO_SOUND_SHOTUZI,
+    SND_UZI_STOP = TR_AUDIO_SOUND_SHOTUZI_END,
+    SND_MAGNUM = TR_AUDIO_SOUND_SHOTMAGNUM,
+    SND_AUTOMAGS = TR_AUDIO_SOUND_SHOTAUTOMAGS,
+    SND_DESERTEAGLE = TR_AUDIO_SOUND_SHOTDESERTEAGLE,
+    SND_REVOLVER = TR_AUDIO_SOUND_SHOTREVOLVER,
+    SND_M16 = TR_AUDIO_SOUND_SHOTM16,
+    SND_M16_STOP = TR_AUDIO_SOUND_SHOTM16_END,
+    SND_MP5 = TR_AUDIO_SOUND_SHOTMP5,
+    SND_ROCKETGUN = TR_AUDIO_SOUND_SHOTROCKETGUN,
+    SND_ROCKETGUN_RELOAD = TR_AUDIO_SOUND_RELOADROCKETGUN,
+    SND_HARPOON_LAND = TR_AUDIO_SOUND_SHOTHARPOON_G,
+    SND_HARPOON_WATER = TR_AUDIO_SOUND_SHOTHARPOON_W,
+    SND_HARPOON_RELOAD_LAND = TR_AUDIO_SOUND_RELOADHARPOON_G,
+    SND_HARPOON_RELOAD_WATER = TR_AUDIO_SOUND_RELOADHARPOON_W,
+    SND_GRENADEGUN = TR_AUDIO_SOUND_SHOTGRENADEGUN,
+    SND_TR2_3_GRENADEGUN_RELOAD = TR_AUDIO_SOUND_TR2_3_RELOADGRENADEGUN,
+    SND_TR2_3_GRENADEGUN_LOCK = TR_AUDIO_SOUND_TR2_3_RELOADGRENADEGUN_LOCK,
+    SND_TR4_C_GRENADEGUN_RELOAD = TR_AUDIO_SOUND_TR4C_RELOADGRENADEGUN,
+    SND_TR4_C_GRENADEGUN_LOCK = TR_AUDIO_SOUND_TR4C_RELOADGRENADEGUN_LOCK,
+    SND_CROSSBOWGUN = TR_AUDIO_SOUND_SHOTCROSSBOW
+};
+
+#define RATE_LOWEST 0.1f
+#define RATE_PISTOL 1.0f
+#define RATE_SHOTGUN 1.0f
+#define RATE_UZI 4.0f
+#define RATE_MAGNUM 0.8f
+#define RATE_AUTOMAGS 1.0f
+#define RATE_DESERTEAGLE 0.8f
+#define RATE_REVOLVER 0.8f
+#define RATE_M16 1.0f
+#define RATE_M16_ALT 1.5f
+#define RATE_MP5 2.0f
+#define RATE_ROCKETGUN 1.0f
+#define RATE_HARPOONGUN 1.0f
+#define RATE_GRENADEGUN 1.0f
+#define RATE_CROSSBOWGUN 1.0f
+#define RATE_GRAPPINGUN 1.0f
+
+#define RANGE_LOWEST 256.0f     // block "1/4"
+#define RANGE_PISTOL 8192.0f
+#define RANGE_SHOTGUN 8192.0f
+#define RANGE_UZI 8192.0f
+#define RANGE_MAGNUM 8192.0f
+#define RANGE_AUTOMAGS 8192.0f
+#define RANGE_DESERTEAGLE 8192.0f
+#define RANGE_REVOLVER 8192.0f
+#define RANGE_M16 16384.0f
+#define RANGE_MP5 16384.0f
+#define RANGE_ROCKET 65535.0f    // explose at impact (no range)
+#define RANGE_GRENADE 8192.0f    // explose at impact (not in TR4+)
+#define RANGE_CROSSBOW 16384.0f  // gravity ?
+#define RANGE_GRAPPLIN 16384.0f  // don't know the range :x
+
+struct weapons_s
+{
+    uint32_t item_id;
+    int ammo_counter;
+    int shot;
+    int draw;
+    int hide;
+    int echo;
+    int reload_1;     // classic reload
+    int reload_2;     // grenadegun reload lock
+    int damage = NULL;

```

Comment:
The 3 preceding fields are not pointers, they should not be initialized using NULL (this triggers a warning when building with gcc).
```suggestion
    int damage = 0;
```

Reply:
#fixed